src/scripts/main.js

```diff
@@ -1,3 +1,27 @@
 'use strict';
 
-// write your code here
+const thead = document.querySelectorAll('thead tr');
+const tbody = document.querySelectorAll('tbody tr');
+const tfoot = document.querySelectorAll('tfoot tr');
+
+const columnIndex = 1;
+
+// eslint-disable-next-line no-shadow
+function insertColumn(rows, columnIndex) {
+  rows.forEach((tr) => {
+    if (tr.children.length > columnIndex) {
+      const th = tr.children[columnIndex];
+      const clonedTh = th.cloneNode(true);
+
+      if (tr.children.length > columnIndex + 2) {
```

Comment:
The condition `tr.children.length > columnIndex + 2` may not handle cases where the number of columns is exactly `columnIndex + 2`. Consider changing the condition to `tr.children.length >= columnIndex + 2` to ensure the cloned column is inserted correctly in all cases.